compiler/src/iree/compiler/Codegen/Common/test/gpu_materialize_encoding_gfx942.mlir

```diff
@@ -1190,3 +1190,70 @@ func.func @batch_matmul_lowering_MFMA_F32_16x16x16_BF16() {
+                                       features = "",
+                                       wgp = <compute =  fp32,
+                                              storage =  b32,
+                                              subgroup =  none,
+                                              dot =  none, mma = [<MFMA_F32_16x16x4_F32>],
+                                              subgroup_size_choices = [64],
+                                              max_workgroup_sizes = [1024, 1024, 1024],
+                                              max_thread_count_per_workgroup = 1024,
+                                              max_workgroup_memory_bytes = 65536,
+                                              max_workgroup_counts = [2147483647, 2147483647, 2147483647],
+                                              max_load_instruction_bits = 128,
+                                              simds_per_wgp = 4,
+                                              vgpr_space_bits = 16384>>,
+  encoding = #iree_gpu.gpu_encoding_layout<configuration = {
+                                              encoding_info = {
+                                                innerDimsPos = [0, 1],
+                                                innerTileSizes = [128, 16],
+                                                outerDimsPerm = [1, 0],
+                                                swizzle = {expandShape = [[["CrossThread", 4 : i16], ["CrossIntrinsic", 8 : i16], ["CrossThread", 4 : i16]],
+                                                                          [["CrossIntrinsic", 4 : i16], ["CrossThread", 4 : i16]]],
+                                                           permutation = [1, 4, 0, 2, 3]}}}>
+}>
+#encoding = #iree_encoding.encoding<operand_index = 0, op_type = matmul, element_types = [f32, f32, f32],
+                                    user_indexing_maps = [affine_map<(d0, d1, d2) -> (d0, d2)>, affine_map<(d0, d1, d2) -> (d2, d1)>, affine_map<(d0, d1, d2) -> (d0, d1)>],
+                                    round_dims_to = array<i64: 32, 32, 32>>
+#pipeline_layout = #hal.pipeline.layout<bindings = [
+  #hal.pipeline.binding<storage_buffer>,
+  #hal.pipeline.binding<storage_buffer>
+]>
+func.func @set_encoding_with_layout() attributes {
+  hal.executable.target = #executable_target_rocm_hsaco_fb
+} {
+  %c0 = arith.constant 0 : index
+  %0 = hal.interface.binding.subspan layout(#pipeline_layout) binding(0) alignment(64) offset(%c0) flags(ReadOnly) : !flow.dispatch.tensor<readonly:tensor<255x513xf32>>
+  %1 = hal.interface.binding.subspan layout(#pipeline_layout) binding(1) alignment(64) offset(%c0) : !flow.dispatch.tensor<writeonly:tensor<255x513xf32, #encoding>>
+  %2 = flow.dispatch.tensor.load %0, offsets = [0, 0], sizes = [255, 513], strides = [1, 1] : !flow.dispatch.tensor<readonly:tensor<255x513xf32>> -> tensor<255x513xf32>
+  %3 = iree_encoding.set_encoding %2 : tensor<255x513xf32> -> tensor<255x513xf32, #encoding>
```

Comment:
In the context, the type converter queries the layout from `getEncodingInfo` using the `#iree_gpu.gpu_encoding_layout<>` attribute.

If we want to make the test be closer to practice, there are two encodings. The encoding on bindings has resolved layout. And the encoding on any other computation ops does not have layouts. E.g.,

https://github.com/iree-org/iree/blob/fb3523be75e441128c8aff3d50c9c929041241ff/compiler/src/iree/compiler/Codegen/Common/test/llvmcpu_materialize_encoding.mlir#L2958-L2978